ChatQnA/kubernetes/manifests/README.md

```diff
@@ -14,63 +23,80 @@ You should have all the images
 
-> [NOTE]
-
-- Be sure to modify HUGGINGFACEHUB_API_TOKEN and other important values in qna_configmap_guadi.yaml and qna_configmap_xeon.yaml
-- Be sure the node has path /mnt/models to store all the models
+```sh
+kubectl create ns chatqa
+kubectl apply -f $(pwd)/chatQnA_xeon.yaml
+```
 
-### Deploy
+**GMC will reconcile chatQnA custom resource and get all related components/services ready**
 
+```sh
+kubectl get service -n chatqa
 ```
-# Deploy Services with Xeon
 
-$ cd ${RepoPath}/ChatQnA/kubernetes/manifests/
-$ ./install_all_xeon.sh
+**Check GMC chatQnA custom resource to get access URL for the pipeline**
+
+```bash
+$kubectl get gmconnectors.gmc.opea.io -n chatqa
+NAME     URL                                                      READY     AGE
+chatqa   http://router-service.chatqa.svc.cluster.local:8080      8/0/8     3m
 ```
 
-### Undeploy
+**Deploy one client pod for testing the chatQnA application**
 
+```bash
+kubectl create deployment client-test -n chatqa --image=python:3.8.13 -- sleep infinity
 ```
-# Remove Services with Xeon
 
-$ ./remove_all_xeon.sh
+**Access the pipeline using the above URL from the client pod**
+
+```bash
+export CLIENT_POD=$(kubectl get pod  -l app=client-test -o jsonpath={.items..metadata.name})
+export accessUrl=$(kubectl get gmc -n chatqa -o jsonpath="{.items[?(@.metadata.name=='chatqa')].status.accessUrl}")
+kubectl exec "$CLIENT_POD" -n chatqa -- curl $accessUrl  -X POST  -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -H 'Content-Type: application/json'
 ```
 
-## Deploy Services with Gaudi
+Should you want to change for instance the LLM model you are using in the ChatQnA pipeline, just edit the custom resource file. For instance, if you want to using Llama-2-7b-chat-hf make the following edit.
 
-> [NOTE]
-> Be sure to modify the all the important value in qna_configmap.yaml
-> Be sure the node has path /mnt/models to store all the models
+**Modify chatQnA custom resource to change to another LLM model**
 
-### Deploy
+```yaml
+- name: Tgi
+  internalService:
+    serviceName: tgi-svc
+    config:
+      LLM_MODEL_ID: Llama-2-7b-chat-hf
+```
 
+Apply the change using
 ```
-$ cd ${RepoPath}/ChatQnA/kubernetes/manifests/
-$ ./install_all_gaudi.sh
+kubectl apply -f $(pwd)/chatQnA_xeon.yaml
 ```
 
-### Undeploy
+**Check the tgi-svc-deployment has been changed to use the new LLM Model**
 
-```
-# Remove Services with Xeon
-$ ./remove_all_gaudi.sh
+```sh
+kubectl get deployment tgi-svc-deployment -n chatqa -o jsonpath="{.spec.template.spec.containers[*].env[?(@.name=='LLM_MODEL_ID')].value}"
 ```
 
-## Verify Services
+**Access the updated pipeline using the above URL from the client pod**
 
+```bash
+kubectl exec "$CLIENT_POD" -n chatqa -- curl $accessUrl  -X POST  -d '{"text":"What is the revenue of Nike in 2023?","parameters":{"max_new_tokens":17, "do_sample": true}}' -H 'Content-Type: application/json'
```

Comment:
one empty space before `-X` and `-d` is recommended.

Reply:
fixed